core/store/src/trie/shard_tries.rs

```diff
@@ -113,43 +113,47 @@ impl ShardTries {
         is_view: bool,
         block_hash: Option<CryptoHash>,
     ) -> Trie {
-        let cache = self.get_trie_cache_for(shard_uid, is_view);
-        // Do not enable prefetching on view caches.
-        // 1) Performance of view calls is not crucial.
-        // 2) A lot of the prefetcher code assumes there is only one "main-thread" per shard active.
-        //    If you want to enable it for view calls, at least make sure they don't share
-        //    the `PrefetchApi` instances with the normal calls.
-        let prefetch_enabled = !is_view && self.0.trie_config.prefetch_enabled();
-        let prefetch_api = prefetch_enabled.then(|| {
-            self.0
-                .prefetchers
-                .write()
-                .expect(POISONED_LOCK_ERR)
-                .entry(shard_uid)
-                .or_insert_with(|| {
-                    PrefetchApi::new(
-                        self.0.store.clone(),
-                        cache.clone(),
-                        shard_uid,
-                        &self.0.trie_config,
-                    )
-                })
-                .0
-                .clone()
-        });
-
-        let storage = Arc::new(TrieCachingStorage::new(
-            self.0.store.clone(),
-            cache,
-            shard_uid,
-            is_view,
-            prefetch_api,
-        ));
-        let flat_storage_chunk_view = block_hash
-            .and_then(|block_hash| self.0.flat_storage_manager.chunk_view(shard_uid, block_hash));
         // Do not use memtries for view queries, for two reasons: memtries do not provide historical state,
         // and also this can introduce lock contention on memtries.
         let memtries = if is_view { None } else { self.get_mem_tries(shard_uid) };
+        let storage: Arc<dyn TrieStorage> = if memtries.is_some() {
+            Arc::new(TrieDBStorage::new(self.0.store.clone(), shard_uid))
```

Comment:
hmm is TrieDBStorage correct here when there is memtrie?

Reply:
Never mind, it seems memtries is not part of the abstraction for TrieStorage so the ShardTries will contain (memtries + TrieDBStorage).